cdc/owner/owner_test.go

```diff
@@ -405,37 +406,53 @@ func (s *ownerSuite) TestHandleJobsDontBlock(c *check.C) {
 	tester.MustUpdate(cdcKey.String(), v)
 
 	// try to add another changefeed
-	changefeedID1 := "test-changefeed1"
-	changefeedInfo1 := &model.ChangeFeedInfo{
+	cf2 := "test-changefeed1"
+	cfInfo2 := &model.ChangeFeedInfo{
 		StartTs: oracle.GoTimeToTS(time.Now()),
 		Config:  config.GetDefaultReplicaConfig(),
+		State:   model.StateNormal,
 	}
-	changefeedStr1, err := changefeedInfo1.Marshal()
+	changefeedStr1, err := cfInfo2.Marshal()
 	c.Assert(err, check.IsNil)
 	cdcKey = etcd.CDCKey{
 		Tp:           etcd.CDCKeyTypeChangefeedInfo,
-		ChangefeedID: changefeedID,
+		ChangefeedID: cf2,
 	}
 	tester.MustUpdate(cdcKey.String(), []byte(changefeedStr1))
 	_, err = owner.Tick(ctx, state)
 	tester.MustApplyPatches()
 	c.Assert(err, check.IsNil)
 	// make sure this changefeed add failed, which means that owner are return
 	// in clusterVersionConsistent check
-	c.Assert(owner.changefeeds[changefeedID1], check.IsNil)
+	c.Assert(owner.changefeeds[cf2], check.IsNil)
 
 	// make sure statusProvider works well
 	ctx1, cancel := context.WithTimeout(context.Background(), time.Second*5)
 	defer cancel()
-	wg := sync.WaitGroup{}
-	wg.Add(1)
+
+	var errIn error
+	var infos map[model.ChangeFeedID]*model.ChangeFeedInfo
+	done := make(chan struct{})
 	go func() {
-		infos, err := statusProvider.GetAllChangeFeedInfo(ctx1)
-		c.Assert(err, check.IsNil)
-		c.Assert(infos[changefeedID], check.NotNil)
-		c.Assert(infos[changefeedID1], check.IsNil)
-		wg.Done()
+		infos, errIn = statusProvider.GetAllChangeFeedInfo(ctx1)
+		done <- struct{}{}
 	}()
-	_, err = owner.Tick(ctx, state)
-	c.Assert(err, check.IsNil)
+
+	ticker := time.NewTicker(20 * time.Millisecond)
+	defer ticker.Stop()
+workloop:
```

Comment:
```suggestion
WorkoLop:
```

See: https://go.dev/doc/effective_go#switch